### PR TITLE
Make building xtensa(UP) on SMP platforms possible

### PR DESCRIPTION
**Pull request by keyonjie**
Add -u option to allow force building xtensa(up) ARCH on platforms which usually building xtensa-smp by default, this is helpful to debug SMP specific issues.

### CODE REVIEW

File: scripts/xtensa-build-all.sh [hunk 1]
```diff
@@ -218,6 +224,12 @@ do
 		HOST="xtensa-cnl-elf"
 		XTENSA_TOOLS_VERSION="RF-2016.4-linux"
 	fi
+	# override default ARCH if BUILD_FORCE_UP is set
+	if [ $BUILD_FORCE_UP == yes ]
```
Comment (xiulipan) [inline scripts/xtensa-build-all.sh]: maybe `if [ "$BUILD_FORCE_UP" == "yes" ]` is safer in bash.

Comment (keyonjie) [reply scripts/xtensa-build-all.sh]: OK, thanks, updated.

**Comment by lgirdwood** [comment]
@keyonjie can you redo this after cmake has been merged.

**Comment by keyonjie** [comment]
> @keyonjie can you redo this after cmake has been merged.

sure.

**Comment by lgirdwood** [comment]
@keyonjie build update merged so you can redo this now.

**Comment by lgirdwood** [comment]
@keyonjie can you check this passes internal CI

**Comment by keyonjie** [comment]
> @keyonjie can you check this passes internal CI

it is just finished, all checks passed.

**Comment by xiulipan** [comment]
@jajanusz 
I did not see cmake take any option about arch 
how to chose "xtensa-smp" or "xtensa"?
@lgirdwood @keyonjie 
I did not see we now use `ARCH` in cmake configure. Please wait for @jajanusz clarify.

**Comment by jajanusz** [comment]
tldr solution:
```
	if [[ "x$BUILD_FORCE_UP" == "xyes" ]]
	then
		echo "CONFIG_SMP=y" > override.config
		make overrideconfig
	fi
```

IMHO we shouldn't keep going for flags, we should just have prepared configs for CIs. I added way for *legacy* support of ci scripts, but it's not something what you usually see in Kconfig based projects.

**Comment by keyonjie** [comment]
> echo "CONFIG_SMP=n" > override.config

Thanks for clarification, transforming to Kconfig is really good, let me update my PR.

**Comment by lgirdwood** [comment]
@jajanusz @xiulipan do you guys now approve ?

**Comment by lgirdwood** [comment]
@keyonjie can you make the > and respond to the CI error.

**Comment by keyonjie** [comment]
> I just noticed it would be better to use `>` instead of `>>`. My mistake, cos i misled you. Anyway it doesn't change final result and is good to merge.

I got your point, as we "make overrideconfig" for every case, so we actually don't need append mode here, right?

**Comment by keyonjie** [comment]
@jajanusz updated to address your question, please help review.

**Comment by jajanusz** [comment]
> > I just noticed it would be better to use `>` instead of `>>`. My mistake, cos i misled you. Anyway it doesn't change final result and is good to merge.
> 
> I got your point, as we "make overrideconfig" for every case, so we actually don't need append mode here, right?

Yea, that's what I mean, you made it even a bit better than what I thought.
Good job.

**Comment by xiulipan** [comment]
LGTM.
